blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/Granularity.java

```diff
@@ -171,29 +171,55 @@ public Range deriveRange(int slot, long referenceMillis) {
     }
 
     /**
+     * Return granularity that maps most closely to requested number of points,
+     * using the algorithm specified in the
+     * {@code GET_BY_POINTS_SELECTION_ALGORITHM} config value. See
+     * {@link #granularityFromPointsInInterval(String, long, long, int, String, long)}.
+     *
+     * @param from beginning of interval (millis)
+     * @param to end of interval (millis)
+     * @param points count of desired data points
+     * @return
+     */
+    public static Granularity granularityFromPointsInInterval(String tenantid, long from, long to, int points) {
+        return granularityFromPointsInInterval(tenantid, from, to, points, GET_BY_POINTS_SELECTION_ALGORITHM, GET_BY_POINTS_ASSUME_INTERVAL, true);
+    }
+    /**
      * Return granularity that maps most closely to requested number of points based on
      * provided selection algorithm
      *
      * @param from beginning of interval (millis)
      * @param to end of interval (millis)
      * @param points count of desired data points
+     * @param algorithm the algorithm to use. Valid values are
+     *                  {@code "GEOMETRIC"}, {@code "LINEAR"}, and
+     *                  {@code "LESSTHANEQUAL"}. Any other value is treated as
+     *                  {@code "GEOMETRIC"}.
+     * @param assumedIntervalMillis FULL resolution is tricky because we don't
+     *                              know the period of check in question.
+     *                              Assume the minimum period and go from there.
+     * @param checkTtl check the specified time against the default ttl values,
+     *                 skipping granularities that don't fit.
      * @return
      */
-    public static Granularity granularityFromPointsInInterval(String tenantid, long from, long to, int points) {
+    public static Granularity granularityFromPointsInInterval(String tenantid, long from, long to, int points, String algorithm, long assumedIntervalMillis, boolean checkTtl) {
```

Comment:
@shintasmith 
The `granularityFromPointsGeometric` method uses the `SafetyTtlProvider.getFinalTTL` method to (I think) get some kind of TTL value. It uses this TTL value to (I think) automatically skip granularities that wouldn't work for the specified `from` time. But that means that the behavior changes based on what the current time is. And that, in turn, means that the tests will fail unless they provide input data based on the current time. The tests I was writing at the time all failed because the method returned `Granularity.MIN_1440`, which didn't make any sense given the inputs. A return value of null or an exception would have been more appropriate. Also, because it was skipping granularities, there were large sections of the method that weren't being executed. It should be noted that this TTL check only happens when the `GEOMETRIC` algorithm is selected. That is, the behavior of the other two algorithms is _different_ in that regard.

I've run into this sort of thing in the past, and it's never made things nice or easy. I've had tests (on other a different project) fail because my machine clock was not properly sync'd to the server's, and there was no way to really know that was the cause. Time-sensitive testing is a losing proposition.

So I decided to provide an overload that allows me to turn off the TTL-checking behavior of the function. That way, I can provide fixed inputs to the method, and confirm that at least the math behind the method is correct. And I don't have to worry about what time it is when I run the tests.

IIRC, the `SafetyTtlProvider` class was also relying on functionality within the `Granularity` class. So trying to test one specific part of `Granularity` was indirectly involving larger chunks of  the class.